src/scanpy/preprocessing/_simple.py

```diff
@@ -632,7 +632,8 @@ def _create_regressor_categorical(
     X: np.ndarray, number_categories: int, cat_array: np.ndarray
 ) -> np.ndarray:
     # create regressor matrix for categorical variables
-    regressors = np.zeros(X.shape, dtype=X.dtype)
+    # would be best to use X dtype but this matches old behavior
+    regressors = np.zeros(X.shape, dtype=np.float32)
```

Comment:
That does seem suboptimal. I wonder how an upgrade path would look here

Reply:
We could easily add an argument, `upcast_regressor` or similar, and tack it onto scanpy 2.0 or something